src/makeInfinityAdder.js

```diff
@@ -4,7 +4,25 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let totalSum = 0;
+
+  function add(num) {
+    if (num !== undefined) {
+      totalSum += num;
+
+      return add;
+    }
+
+    if (num === undefined) {
+      const newSum = totalSum;
+
+      totalSum = 0;
+
+      return newSum;
+    }
```

Comment:
The check for `num === undefined` is redundant since you are already returning `add` if `num` is not undefined. You can simplify the function by removing the second `if` statement and directly returning the sum when no argument is passed.